### ISSUE TITLE
Unable to get as good result as in torch

### ISSUE THREAD

**Issue by ruotianluo** [junyanz/pytorch-CycleGAN-and-pix2pix#5]
Hi,
I ran your code on horse2zebra, with loadSize, fineSize and which_model_netG changed to match torch version. However, I couldn't get good results.

Your pytorch model here do have some differences to torch, like different padding, and different training strategy. 

I'm wondering if you see similar problems.

**Comment by junyanz** [comment]
Hi Ruotian, 

Thanks for reporting it. We are looking at it as well. I tested the model on maps before, and results are just slightly worse than those generated by Torch version. The default setting also doesn't use the reply image buffer, which might make a difference. This is a work in progress, and we are working on improving PyTorch results.   Let us know if you find any issues in the code.

Best,
Jun-Yan

**Comment by griegler** [comment]
Hi,
You call `NLayerDiscriminator` with `use_sigmoid` in networks.py, but it is never used in the constructor of the class. Hence, you always add a sigmoid output unit. Might this be the problem?
Best, 
Gernot

**Comment by taesung89** [comment]
Thank you so much @griegler! This is definitely a bug that should hurt the performance. 

We will test if this bug fixes the quality and close the issue.

**Comment by junyanz** [comment]
We fixed a few small issues. Now the cyclegan results on maps and pix2pix results on facades should be comparable to those generated by Torch implementation. We are testing more datasets now.

**Comment by petergerten** [comment]
Did you test the horse2zebra dataset yet? Also do you intend to share your pretrained model? I would really like to compare as my results do not look good at all.

http://imgur.com/a/yWuGF

**Comment by taesung89** [comment]
Here's my result of test images on Torch/Pytorch. They look similar. 

https://ibb.co/jYm3AQ

@petergerten, I had similar result as yours when we had affine=True as the default option in our Instance Normalization code. Setting affine=False seemed to improve the result. Our Torch implementation uses affine=False. 

The image below shows the difference between affine=True and False. It is subtle but noticeable. 

From left, columns are input horse, affine=True, affine=False, input zebra, affine=True, affine=False. 

https://ibb.co/cRUjH5

Does this look similar to your situation? We updated the codebase on Jul 3rd (233630e79d79901faff420eb0ae481b35d952f9). Have you run your code after this commit?
